reference/latest/src/main/java/com/example/docs/item/ModItems.java

```diff
@@ -0,0 +1,126 @@
+package com.example.docs.item;
+
+import net.minecraft.entity.effect.StatusEffectInstance;
+import net.minecraft.entity.effect.StatusEffects;
+import net.minecraft.item.ArmorItem;
+import net.minecraft.item.FoodComponent;
+import net.minecraft.item.Item;
+import net.minecraft.item.ItemGroups;
+import net.minecraft.item.SwordItem;
+import net.minecraft.registry.Registries;
+import net.minecraft.registry.Registry;
+import net.minecraft.registry.RegistryKey;
+import net.minecraft.util.Identifier;
+
+import net.fabricmc.fabric.api.item.v1.FabricItemSettings;
+import net.fabricmc.fabric.api.itemgroup.v1.ItemGroupEvents;
+import net.fabricmc.fabric.api.registry.CompostingChanceRegistry;
+import net.fabricmc.fabric.api.registry.FuelRegistry;
+
+import com.example.docs.FabricDocsReference;
+import com.example.docs.item.armor.GuiditeArmorMaterial;
+import com.example.docs.item.custom.LightningStick;
+import com.example.docs.item.tool.GuiditeMaterial;
+
+// :::1
+public class ModItems {
+	// :::1
+
+	// :::6
+	public static final Item GUIDITE_HELMET = register(new ArmorItem(GuiditeArmorMaterial.INSTANCE, ArmorItem.Type.HELMET, new Item.Settings()), "guidite_helmet");
+	public static final Item GUIDITE_BOOTS = register(new ArmorItem(GuiditeArmorMaterial.INSTANCE, ArmorItem.Type.BOOTS, new Item.Settings()), "guidite_boots");
+	public static final Item GUIDITE_LEGGINGS = register(new ArmorItem(GuiditeArmorMaterial.INSTANCE, ArmorItem.Type.LEGGINGS, new Item.Settings()), "guidite_leggings");
+	public static final Item GUIDITE_CHESTPLATE = register(new ArmorItem(GuiditeArmorMaterial.INSTANCE, ArmorItem.Type.CHESTPLATE, new Item.Settings()), "guidite_chestplate");
+	// :::6
+	public static final Item LIGHTNING_STICK = register(new LightningStick(new FabricItemSettings()), "lightning_stick");
+	// :::7
+	public static final Item GUIDITE_SWORD = register(new SwordItem(GuiditeMaterial.INSTANCE, 2, 0.5F, new FabricItemSettings()), "guidite_sword");
+	// :::7
+
+	// :::5
+	public static final FoodComponent SUSPICIOUS_FOOD_COMPONENT = new FoodComponent.Builder()
+			.alwaysEdible()
+			.snack()
+			// The duration is in ticks, 20 ticks = 1 second
+			.statusEffect(new StatusEffectInstance(StatusEffects.POISON, 6 * 20, 1), 1.0f)
+			.build();
+	// :::5
+
+	// :::2
+	public static final Item SUSPICIOUS_SUBSTANCE = register(
+			// Ignore the food component for now, we'll cover it later in the food section.
+			new Item(new FabricItemSettings().food(SUSPICIOUS_FOOD_COMPONENT)),
+			"suspicious_substance"
+	);
+	// :::2
+
+	// :::1
+	// We can use generics (<T extends Item>), so we don't need to
```

Comment:
I dont think the generics provide anything beyond confusion. This can take an `Item` and return an `Item`.

I also think we should try to standardise how things are registered. In `FabricDocsReferenceItems` the item group is registered in `onInitialize` but here its done statically.